haystack/document_stores/elasticsearch8.py

```diff
@@ -0,0 +1,335 @@
+
+        self._RequestError = RequestError
+
+    @classmethod
+    def _init_elastic_client(
+        cls,
+        host: Union[str, List[str]],
+        port: Union[int, List[int]],
+        username: str,
+        password: str,
+        api_key_id: Optional[str],
+        api_key: Optional[str],
+        scheme: str,
+        ca_certs: Optional[str],
+        verify_certs: bool,
+        timeout: int,
+        use_system_proxy: bool,
+    ) -> Elasticsearch:
+        hosts = cls._prepare_hosts(host, port, scheme)
+
+        if (api_key or api_key_id) and not (api_key and api_key_id):
+            raise ValueError("You must provide either both or none of `api_key_id` and `api_key`.")
+
+        node_class: Type[BaseNode] = Urllib3HttpNode
+        if use_system_proxy:
+            node_class = RequestsHttpNode
+
+        if api_key_id and api_key:
+            # api key authentication
+            if ca_certs is not None:
+                client = Elasticsearch(
+                    hosts=hosts,
+                    api_key=(api_key_id, api_key),
+                    ca_certs=ca_certs,
+                    verify_certs=verify_certs,
+                    request_timeout=timeout,
+                    node_class=node_class,
+                )
+            else:
+                client = Elasticsearch(
+                    hosts=hosts,
+                    api_key=(api_key_id, api_key),
+                    verify_certs=verify_certs,
+                    request_timeout=timeout,
+                    node_class=node_class,
+                )
+        elif username:
+            # standard http_auth
+            if ca_certs is not None:
+                client = Elasticsearch(
+                    hosts=hosts,
+                    basic_auth=(username, password),
+                    ca_certs=ca_certs,
+                    verify_certs=verify_certs,
+                    request_timeout=timeout,
+                    node_class=node_class,
+                )
+            else:
+                client = Elasticsearch(
+                    hosts=hosts,
+                    basic_auth=(username, password),
+                    verify_certs=verify_certs,
+                    request_timeout=timeout,
+                    node_class=node_class,
+                )
+        else:
+            # there is no authentication for this elasticsearch instance
+            if ca_certs is not None:
+                client = Elasticsearch(
+                    hosts=hosts,
+                    ca_certs=ca_certs,
+                    verify_certs=verify_certs,
+                    request_timeout=timeout,
+                    node_class=node_class,
+                )
+            else:
+                client = Elasticsearch(
+                    hosts=hosts,
+                    basic_auth=(username, password),
+                    verify_certs=verify_certs,
+                    request_timeout=timeout,
+                    node_class=node_class,
+                )
```

Comment:
Let's keep the logic simple by inverting the check. It's easier to reason about these kind of cases when they're straightforward.

```suggestion
            if ca_certs:
                client = Elasticsearch(
                    hosts=hosts,
                    verify_certs=verify_certs,
                    request_timeout=timeout,
                    node_class=node_class,
                )
            else:
                client = Elasticsearch(
                    hosts=hosts,
                    ca_certs=ca_certs,
                    basic_auth=(username, password),
                    verify_certs=verify_certs,
                    request_timeout=timeout,
                    node_class=node_class,
                )
```

Same thing for the above `if`s.